src/NodeParser/TypeReferenceNodeParser.ts

```diff
@@ -20,41 +21,60 @@ export class TypeReferenceNodeParser implements SubNodeParser {
     }
 
     public createType(node: ts.TypeReferenceNode, context: Context): BaseType {
-        const typeSymbol = this.typeChecker.getSymbolAtLocation(node.typeName)!;
+        const typeSymbol =
+            this.typeChecker.getSymbolAtLocation(node.typeName) ??
+            // When the node doesn't have a valid source file, its position is -1, so we can't
+            // search for a symbol based on its location. In that case, the ts.factory defines a symbol
+            // property on the node itself.
+            (node.typeName as unknown as ts.Type).symbol;
+
+        // Wraps promise type to avoid resolving to a empty Object type.
+        if (typeSymbol.name === "Promise") {
+            return this.childNodeParser.createType(node.typeArguments![0]!, this.createSubContext(node, context));
```

Comment:
Should the subcontext have the promise inner or outer type node?

```suggestion
            return this.childNodeParser.createType(node.typeArguments![0]!, this.createSubContext(node.typeArguments![0]!, context));
```

Reply:
Is this a question for me? I don't know. What is the impact on the schema?